hamocc/mo_hamocc4bcm.F90

```diff
@@ -0,0 +1,397 @@
+    !    related code-restructuring
+    !  - added sediment bypass preprocessor option
+    !  J.Schwinger,      *NORCE Climate, Bergen*   2020-05-28
+    !  - restructuring of iHAMOCC code, cleanup parameter list
+    !  - boundary conditions (dust, riverinput, N-deposition) are now passed as
+    !    an argument
+    !******************************************************************************
+
+    use mod_xc,           only: mnproc
+    use mo_carbch,        only: atmflx,ocetra,atm,&
+                                atm_cfc11_nh,atm_cfc11_sh,atm_cfc12_nh,atm_cfc12_sh,atm_sf6_nh,atm_sf6_sh
+    use mo_biomod,        only: strahl
+    use mo_control_bgc,   only: ldtrunbgc,dtbgc,ldtbgc,io_stdo_bgc,dtbgc,ndtdaybgc, &
+                                do_sedspinup,sedspin_yr_s,sedspin_yr_e,sedspin_ncyc, &
+                                use_BROMO, use_CFC, use_PBGC_CK_TIMESTEP,&
+                                use_BOXATM, use_sedbypass,ocn_co2_type
+    use mo_param1_bgc,    only: iatmco2,iatmdms,nocetra,nriv,iatmbromo
+    use mo_vgrid,         only: set_vgrid
+    use mo_apply_fedep,   only: apply_fedep
+    use mo_apply_rivin,   only: apply_rivin
+    use mo_apply_ndep,    only: apply_ndep
+    use mo_apply_oafx,    only: apply_oafx
+    use mo_boxatm,        only: update_boxatm
+    use mo_inventory_bgc, only: inventory_bgc
+    use mo_sedshi,        only: sedshi
+    use mo_get_cfc,       only: get_cfc
+    use mo_powach,        only: powach
+    use mo_preftrc,       only: preftrc
+    use mo_cyano,         only: cyano
+    use mo_ocprod,        only: ocprod
+    use mo_carchm,        only: carchm
+
+    ! Arguments
+    integer, intent(in)    :: kpie                                            ! 1st dimension of model grid.
+    integer, intent(in)    :: kpje                                            ! 2nd dimension of model grid.
+    integer, intent(in)    :: kpke                                            ! 3rd (vertical) dimension of model grid.
+    integer, intent(in)    :: kbnd                                            ! number of halo grid points.
+    integer, intent(in)    :: kplyear                                         ! current year.
+    integer, intent(in)    :: kplmon                                          ! current month.
+    integer, intent(in)    :: kplday                                          ! current day.
+    integer, intent(in)    :: kldtday                                         ! number of time step in current day.
+    real,    intent(in)    :: pdlxp  (kpie,kpje)                              ! size of grid cell (longitudinal) [m].
+    real,    intent(in)    :: pdlyp  (kpie,kpje)                              ! size of grid cell (latitudinal) [m].
+    real,    intent(in)    :: pddpo  (kpie,kpje,kpke)                         ! size of grid cell (depth) [m].
+    real,    intent(in)    :: prho   (kpie,kpje,kpke)                         ! density [kg/m^3].
```

Comment:
Dear @mvertens , the unit for density `prho` could be fixed in the comment (formerly wrong): `g/cm^3`

Reply:
Done.